frontend/notification/pushNotificationService.js

```diff
@@ -124,13 +147,22 @@
         };
 
         /**
-         * Check if the user has already conceded the permission
-         * using Notification object.
+         * Look for firebase object, in firebase array, corresponding to the current device token.
+         * @returns {Promise<FirebaseObject|undefined>}
          * @private
          */
-        service._hasNotificationPermission = function hasNotificationPermission() {
-            const { permission } = Notification;
-            return permission === "granted";
+        service._getTokenObjectInFirebaseArray = () => {
+             return messaging && messaging.getToken().then((deviceToken) => {
+                return service.firebaseArrayNotifications.$loaded().then((updatedArray) => {
+                    let tokenObject;
+                    updatedArray.map((objectToken) => {
```

Comment:
the [find](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/find) function does that more efficiently